src/BoundaryConditions/fill_halo_regions_open.jl

```diff
@@ -1,49 +1,43 @@
 function fill_bounded_wall_normal_halo_regions!(velocities, clock, fields)
     grid = velocities.u.grid
 
-    TX, TY, TZ = topology(grid)
-
-    if TX() isa Bounded
-        fill_u_open_halo_regions!(velocities.u, grid, clock, fields)
-    end
-
-    if TY() isa Bounded
-        fill_v_open_halo_regions!(velocities.v, grid, clock, fields)
-    end
-
-    if TZ() isa Bounded
-        fill_w_open_halo_regions!(velocities.w, grid, clock, fields)
-    end
+    fill_open_halo_regions!(velocities.u, grid, clock, fields)
+    fill_open_halo_regions!(velocities.v, grid, clock, fields)
+    fill_open_halo_regions!(velocities.w, grid, clock, fields)
 
     return nothing
 end
 
-function fill_u_open_halo_regions!(u, grid, args...)
-    arch = architecture(grid)
-
-    west_bc = u.boundary_conditions.west
-    east_bc = u.boundary_conditions.east
+const bounded_grid = Union{AbstractGrid{Any, Bounded}, AbstractGrid{Any, Any, Bounded}, AbstractGrid{Any, Any, Any, Bounded}}
 
-    launch!(arch, grid, :yz, _fill_west_and_east_open_halo!, u, west_bc, east_bc, instantiated_location(u), grid, args)
-end
+@inline fill_open_halo_regions!(field, grid, args...) = nothing
 
-function fill_v_open_halo_regions!(v, grid, args...)
+function fill_open_halo_regions!(field, grid::bounded_grid, loc, args...)
```

Comment:
type aliases (and types) use TitleCase

Reply:
I'll have to read this code more closely, but I would think you need `XBoundedGrid`, `YBoundedGrid`, and `ZBoundedGrid` for this, since the special case is `Field{Face, Center, Center}` + `XBoundedGrid`, etc.

Those type aliases are defined in the `Grids` module.

Reply:
I think these alias' also include left/right connected grids which I'm not sure we want to support here?

https://github.com/CliMA/Oceananigans.jl/blob/ddf322f7faa47980bfe4de5bbd203696f419f9e2/src/Grids/inactive_node.jl#L38-L47

But I'm also not sure what connected grids are meant for.

Reply:
Hmm. `LeftConnected` is bounded on the right and connected to another node on the left (eg when using MPI). We do want to support that right?